lib/Jsrt/ChakraDebug.h

```diff
@@ -0,0 +1,567 @@
+        /// </summary>
+        JsDiagDebugEventDebuggerStatement = 4,
+        /// <summary>
+        ///     Indicates a break due to async break.
+        /// </summary>
+        JsDiagDebugEventAsyncBreak = 5,
+        /// <summary>
+        ///     Indicates a break due to a runtime script exception.
+        /// </summary>
+        JsDiagDebugEventRuntimeException = 6
+    } JsDiagDebugEvent;
+
+    /// <summary>
+    ///     Break on Exception attributes.
+    /// </summary>
+    typedef enum _JsDiagBreakOnExceptionAttributes
+    {
+        /// <summary>
+        ///     Don't break on any exception.
+        /// </summary>
+        JsDiagBreakOnExceptionAttributeNone = 0x0,
+        /// <summary>
+        ///     Break on uncaught exception.
+        /// </summary>
+        JsDiagBreakOnExceptionAttributeUncaught = 0x1,
+        /// <summary>
+        ///     Break on first chance exception.
+        /// </summary>
+        JsDiagBreakOnExceptionAttributeFirstChance = 0x2
+    } JsDiagBreakOnExceptionAttributes;
+
+    /// <summary>
+    ///     Stepping types.
+    /// </summary>
+    typedef enum _JsDiagStepType
+    {
+        /// <summary>
+        ///     Perform a step operation to next statement.
+        /// </summary>
+        JsDiagStepTypeStepIn = 0,
+        /// <summary>
+        ///     Perform a step out from the current function.
+        /// </summary>
+        JsDiagStepTypeStepOut = 1,
+        /// <summary>
+        ///     Perform a single step over after a debug break if the next statement is a function call, else behaves as a stepin.
+        /// </summary>
+        JsDiagStepTypeStepOver = 2
+    } JsDiagStepType;
+
+    /// <summary>
+    ///     User implemented callback routine for debug events.
+    /// </summary>
+    /// <remarks>
+    ///     Use <c>JsDiagStartDebugging</c> to register the callback.
+    /// </remarks>
+    /// <param name="debugEvent">The type of JsDiagDebugEvent event.</param>
+    /// <param name="eventData">Additional data related to the debug event.</param>
+    /// <param name="callbackState">The state passed to <c>JsDiagStartDebugging</c>.</param>
+    typedef void (CHAKRA_CALLBACK * JsDiagDebugEventCallback)(_In_ JsDiagDebugEvent debugEvent, _In_ JsValueRef eventData, _In_opt_ void* callbackState);
+
+    /// <summary>
+    ///     Starts debugging in the given runtime.
+    /// </summary>
+    /// <param name="runtimeHandle">Runtime to put into debug mode.</param>
+    /// <param name="debugEventCallback">Registers a callback to be called on every JsDiagDebugEvent.</param>
+    /// <param name="callbackState">User provided state that will be passed back to the callback.</param>
+    /// <returns>
+    ///     The code <c>JsNoError</c> if the operation succeeded, a failure code otherwise.
+    /// </returns>
+    /// <remarks>
+    ///     The runtime should be active on the current thread and should not be in debug state
```

Comment:
`.` after sentence. Add comments that runtime handle passed in must be the current runtime.